src/components/ProductCard/index.tsx

```diff
@@ -0,0 +1,20 @@
+import React from 'react'
+import Link from 'next/link'
+
+const ProductCard = () => {
+  return (
+    <div className="product-card">
+      <Link href="/product-view">
+        <a className="product-card__link">
+          <div className="product-card__content">
+            <span>
+              Click Me!
+            </span>
+          </div>
+        </a>
+      </Link>
+    </div>
+  )
+}
```

Comment:
1. Since we are going to pass some data such as id when routing to another page. I would consider using `Next Router` instead.
2. Do you think we should use this path name ('/product-view'). How about making it more unique by adding product id instead ('/products/{id}')
3. When there is only a small text, I personally like to align it in one line (`<span>Click Me!</span>`) instead. What do you think?
```suggestion
interface ProductCard {
  id: string
  // TODO: Other props will be added
}

const ProductCard = ({ id }: ProductCard) => {
  const router = useRouter()

  const handleProductCardClick = () => {
    router.push(`/products/${id}`)
  }

  return (
    <div className="product-card">
      <a className="product-card__link" onClick={handleProductCardClick}>
        <div className="product-card__content">
          <span>Click Me!</span>
        </div>
      </a>
    </div>
  )
}
```